PostHog/Replay/PostHogDefaultHTTPProtocol.swift

```diff
@@ -0,0 +1,281 @@
+                //    - Need to figure out a way to use custom sessions and configurations here
+                session = URLSession(configuration: .default, delegate: self, delegateQueue: nil)
+            }
+        }
+
+        override public class func canInit(with request: URLRequest) -> Bool {
+            canHandle(request: request)
+        }
+
+        override class func canInit(with task: URLSessionTask) -> Bool {
+            canHandle(task: task)
+        }
+
+        private static func canHandle(task: URLSessionTask) -> Bool {
+            if #available(iOS 13.0, macOS 10.15, *) {
+                // No ws support for now
+                if task is URLSessionWebSocketTask {
+                    return false
+                }
+            }
+
+            guard let request = task.currentRequest else { return false }
+            return canHandle(request: request)
+        }
+
+        private static func canHandle(request: URLRequest) -> Bool {
+            guard PostHogSDK.shared.isCaptureNetworkTelemetryEnabled() else { return false }
+
+            guard shouldHandleHost(request) else { return false }
+
+            // check if this request has already been handled
+            guard !isHandling(request: request) else { return false }
+
+            return true
+        }
+
+        private class func shouldHandleHost(_: URLRequest) -> Bool {
+            // just a placeholder for now, we could potentially choose to ignore/allow some hosts from config
+            true
+        }
+
+        override public func startLoading() {
+            // mark as handled
+            let request = Self.markHandling(request: request)
+            // collect info
+            currentSample.start(request: request)
+            // execute
+            sessionDataTask = session?.dataTask(with: request)
+            sessionDataTask?.resume()
+        }
+
+        override public func stopLoading() {
+            currentSample.stop()
+            sessionDataTask?.cancel()
+            session?.invalidateAndCancel()
+        }
+
+        override public class func canonicalRequest(for request: URLRequest) -> URLRequest {
+            request
+        }
+
+        deinit {
+            session = nil
+            sessionDataTask = nil
+        }
+
+        private func processCurrentSample() {
+            var snapshotsData: [Any] = []
+            let requestsData = [currentSample.toDict()]
+            let payloadData: [String: Any] = [
+                "requests": requestsData,
+            ]
+            let pluginData: [String: Any] = [
+                "plugin": "rrweb/network@1",
+                "payload": payloadData,
+            ]
+
+            let data: [String: Any] = [
+                "type": 6,
+                "data": pluginData,
+                "timestamp": currentSample.timestamp,
+            ]
+
+            snapshotsData.append(data)
+
+            PostHogSDK.shared.capture("$snapshot", properties: [
```

Comment:
we need to check `isCaptureNetworkEnabled` before processing a request and before capturing so we are sure we don't capture when there's no active session